### ISSUE TITLE
table, tr, td element with role=table, row, cell should be categorized as Recommendations and not Violation

### ISSUE THREAD

**Issue by dehru** [IBMa/equal-access#2338]
The IBM Equal Access Accessibility Checker is categorizing the redundant roles defined for table, tr, td as Violations.  However those roles are allowed in the spec, but they are just not recommended.  This is stated in the [ARIA In HTML](https://www.w3.org/TR/html-aria/#avoid-specifying-redundant-roles)  publication updated July 23, 2025.  Specifically in the [reference table in section 4](https://www.w3.org/TR/html-aria/#docconformance).

The following structure generates a violation for each instance in the table.

```
<table role="table">
  <tr role="row">
    <td role="cell"></td>
  </tr>
</table>
```

**To Reproduce**
1. Go to 'https://v16.material.angular.dev/components/table/overview#table-basic'
2. Run the accessibility checker on the example table.
3. See that 250 violations are reported on table structure and redundant use of role.

**Expected behavior**
Because the ARIA in HTML document published by w3c says that these roles are allowed, but not recommended, I expect that this should be categorized as Recommendations or Needs Review.  It should not be categorized as a Violation.

**Screenshots**

<img width="1362" height="778" alt="Image" src="https://github.com/user-attachments/assets/4f71d7a9-74f8-41b6-892c-8e15429db7e1" />
<img width="1824" height="657" alt="Image" src="https://github.com/user-attachments/assets/d0e16357-7bdc-4434-878f-25f3aae85819" />

**Desktop:**
 - OS: Ios
 - Browser: chrome
 - Version: 138.0.7204.169
 - IBM Equal Accesses Extension Version: 4.0.6.9999

**Additional context**
Because we have data heavy content that utilizes tables quite a bit, we get a Violation for every single cell in every table.  Additionally, we are using [Angular Material Components](https://v16.material.angular.dev/), which provide this table that we use out of the box.  They have referenced this in their [accessibility notes](https://v16.material.angular.dev/components/table/overview#accessibility) in their component documentation.

## Definition of Done

- [ ] Design artifacts reviewed
- [ ] Work item implemented
- [ ] Automated test case created (when applicable)
- [ ] Unit test completed
- [ ] Run Checker (when applicable)
- [ ] Manual and Screen Reader Testing (when applicable)
- [ ] Video recording of user experience walkthrough (if impact to UX)
- [ ] PR created

**Comment by shunguoy** [comment]
triage: should be downgraded to 'Recommendation", the message needs to be updated to be consistent.

**Comment by shunguoy** [comment]
Correction: ignore the case when a redundant rule is used for a table, tr and td elements for the rule, which is already covered under the aria_role_redundant rule.
